### PR TITLE
[geometry] Fix Meshcat browser console error messages

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Prior to this, `Uncaught ReferenceError: urlParams is not defined` was printed from any static HTML page.

Amends #20430.

+@SeanCurtis-TRI for both reviews, please.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)

### CODE REVIEW

**Review by SeanCurtis-TRI — COMMENTED** [review]
Alternative solution suggested. See below.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-:-NoSNOpHEvVV2pFBoS0I:bctankg)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSMk2z1q-GlH52Zrx8:b-vg30eh) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
> ```HTML
>     }
> 
>     const queryString = window.location.search;
> ```

I'd suggest the real solution is to move these two lines up above `<!-- CONNECTION BLOCK BEGIN -->` (line 138).

That way the static html can still be passed the webxr parameters and webxr can be used even on a static page.  In fact, the trailing webxr initializing code could move up with it as well so the connection boiler plate would simply be the last chunk of the page's javascript.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSOJBD-m_2VjmJV1jv:b-zdkdbw) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

I'd suggest the real solution is to move these two lines up above `<!-- CONNECTION BLOCK BEGIN -->` (line 138).

That way the static html can still be passed the webxr parameters and webxr can be used even on a static page.  In fact, the trailing webxr initializing code could move up with it as well so the connection boiler plate would simply be the last chunk of the page's javascript.
</blockquote></details>

Maybe you'd like to do that in a follow-up PR?

Promising support for offline VR/XR also means adding more test scripting to `meshcat_manual_test` to ensure that the feature continues to work, and I'm not sure that's worth it if we think the long-term goal is to offer a UI settings panel in lieu of the query params?

**Review by SeanCurtis-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSPPMgDk-paa4TEQ8M:buazlt1) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Maybe you'd like to do that in a follow-up PR?

Promising support for offline VR/XR also means adding more test scripting to `meshcat_manual_test` to ensure that the feature continues to work, and I'm not sure that's worth it if we think the long-term goal is to offer a UI settings panel in lieu of the query params?
</blockquote></details>

It's *only* an offline thing.

The error is the implication that it's part of communication with a server. I think that's orthogonal to what we might do with future UX.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSPtFX9OVXCusn4Bvx:bm5gdwv) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

It's *only* an offline thing.

The error is the implication that it's part of communication with a server. I think that's orthogonal to what we might do with future UX.
</blockquote></details>

Maybe I wasn't clear?  The setting panel thing wasn't a load-bearing part of my discussion, maybe I should have omitted that part.

To recap: Offline VR/XR is not currently supported.  It doesn't work, and it has never worked.  One side-effect of that brokenness is browser console spam.  This PR fixes that spam, so is a tiny monotonic improvement.

Adding offline VR support would be a new feature that requires new tests.  I don't want to write those tests, because neither I nor my relevant users need that feature.  Adding that feature & tests would distract from more important work.  (The current use case for VR is teleop, which doesn't make sense when offline.)

**Review by SeanCurtis-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform)

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSRVBS1WjkImweScIk:b-qg7sdt) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Maybe I wasn't clear?  The setting panel thing wasn't a load-bearing part of my discussion, maybe I should have omitted that part.

To recap: Offline VR/XR is not currently supported.  It doesn't work, and it has never worked.  One side-effect of that brokenness is browser console spam.  This PR fixes that spam, so is a tiny monotonic improvement.

Adding offline VR support would be a new feature that requires new tests.  I don't want to write those tests, because neither I nor my relevant users need that feature.  Adding that feature & tests would distract from more important work.  (The current use case for VR is teleop, which doesn't make sense when offline.)
</blockquote></details>

I can appreciate your point that it hasn't been tested. I'd argue it hasn't worked because of an oversight in the original PR and not by design. I can accept the idea that to offer it as a proper feature, it needs to be tested (even as horrible a medium as `meshcat_manual_test` is).

The PR clearly placed the VR stuff outside of the static block. That is clear intention. The fact that it went untested was a failure in the PR process. No one was thinking about the html mangling applied by the static html process.

If that PR *had* been thinking about static html, the additional code would've likewise been placed outside the communication block, the feature would've been available for static html, it would have been equally untested, but you wouldn't be proposing this change.

This PR's patch doesn't fix the defect in the original PR. This merely papers over a symptom. In fact, it swaps one defect for another. By placing this code within the communication block, it implies a relationship that wasn't intended and doesn't actually exist.

Your final point is its not worth your time to fix it for real. I can accept that. Lots of defects exist and they require priority. But the defect transformation has moved from one easily observable defect (about which no one else has ever complained) for a different defect (very difficult to detect). The only documentation is *this* discussion. At the very least, I feel you're obliged to put in a TODO or, better yet, an issue.

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform), needs at least two assigned reviewers

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSXIk819v64H-vS4Sw:bstgnez) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
Ah, I see your perspective now.

To me, this was never intended to be a working feature.
To you, it was supposed to be working, but had a bug.
The docs aren't clear one way or another.

I've squared that away by clarifying the docs to say it's not supported, along with a hint in the HTML.

> The PR clearly placed the VR stuff outside of the static block. That is clear intention... By placing this code within the communication block, it implies a relationship that wasn't intended and doesn't actually exist.

I didn't read it that way.  Those PRs were a big pile of confusion.  Some of it leaked into the code and never got wiped up.  Since nobody ever tried using offline xr, I think it's fair to decide that it wasn't supposed to be working.

The code is self-consistent again now -- offline mode is future work.

> (about which no one else has ever complained)

That indicates a hole in our testing.  I've amended the manual test now to monitor browser console output.

**Review by SeanCurtis-TRI — COMMENTED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791)* status: 1 unresolved discussion, LGTM missing from assignee SeanCurtis-TRI(platform), needs at least two assigned reviewers

___
*[`geometry/meshcat.html` line 177 at r1](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSMk2z1q-GlH52Zrx7:-NoSs8Tl3W8vpREYIIyB:b-ut2lz3) ([raw file](https://github.com/RobotLocomotion/drake/blob/892a9c15cfc733e0f0187ec610ff78080dbd1df8/geometry/meshcat.html#L177)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Ah, I see your perspective now.

To me, this was never intended to be a working feature.
To you, it was supposed to be working, but had a bug.
The docs aren't clear one way or another.

I've squared that away by clarifying the docs to say it's not supported, along with a hint in the HTML.

> The PR clearly placed the VR stuff outside of the static block. That is clear intention... By placing this code within the communication block, it implies a relationship that wasn't intended and doesn't actually exist.

I didn't read it that way.  Those PRs were a big pile of confusion.  Some of it leaked into the code and never got wiped up.  Since nobody ever tried using offline xr, I think it's fair to decide that it wasn't supposed to be working.

The code is self-consistent again now -- offline mode is future work.

> (about which no one else has ever complained)

That indicates a hole in our testing.  I've amended the manual test now to monitor browser console output.
</blockquote></details>

The changes you've made make it clear that we understand one another. I'm perfectly content with this resolution.
___
*[`geometry/test/meshcat_manual_test.cc` line 207 at r2](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-NoSsInO4HHNDUJdhnJI:-NoSsInO4HHNDUJdhnJJ:bkuskko) ([raw file](https://github.com/RobotLocomotion/drake/blob/b4d64b59a65b3c56ed925254979ca820af4518b1/geometry/test/meshcat_manual_test.cc#L207)):*
> ```cpp
> 
>   std::cout << R"""(
> Open the developer tools of your browser (F12) and within that panel switch to
> ```

BTW I like this addition the most.

**Comment by jwnimmer-tri** [comment]
+@sammy-tri for platform review per schedule, please.

**Review by sammy-tri — COMMENTED** [review]
:lgtm: (confirmed the error is gone)

> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-:-NoTRPdY5nnaBS1Xde7a:bnwgsnp)* status: LGTM missing from assignee SeanCurtis-TRI(platform)

**Comment by jwnimmer-tri** [comment]
@SeanCurtis-TRI ping for LGTM emoji?

**Review by SeanCurtis-TRI — COMMENTED** [review]
Sorry about that. I got distracted.  :LGTM:

> *[Reviewable](https://reviewable.io/reviews/RobotLocomotion/drake/20791#-:-NoTXE_c1SejERfTca9H:bxsn246)* status: <span>:shipit:</span> complete! all discussions resolved, LGTM from assignees sammy-tri(platform),SeanCurtis-TRI(platform)
